src/clib/pio_getput_int.c

```diff
@@ -473,24 +473,24 @@ int PIOc_put_vars_tc(int ncid, int varid, const PIO_Offset *start, const PIO_Off
 
             /* Get the dimids for this var. */
             if ((ierr = PIOc_inq_vardimid(ncid, varid, dimid)))
-                return check_netcdf(file, ierr, __FILE__, __LINE__);
+                return check_netcdf(ios, file, ierr, __FILE__, __LINE__);
 
             /* Get the length of each dimension. */
             for (int vd = 0; vd < ndims; vd++)
             {
                 if ((ierr = PIOc_inq_dimlen(ncid, dimid[vd], &dimlen[vd])))
-                    return check_netcdf(file, ierr, __FILE__, __LINE__);
+                    return check_netcdf(ios, file, ierr, __FILE__, __LINE__);
                 LOG((3, "dimlen[%d] = %d", vd, dimlen[vd]));
             }
         }
 
         /* Allocate memory for these arrays, now that we know ndims. */
         if (!(rstart = malloc(ndims * sizeof(PIO_Offset))))
-            return check_netcdf(file, PIO_ENOMEM, __FILE__, __LINE__);
```

Comment:
That is a good point. Let me ponder that. It would look more natural to use pio_err() for malloc errors. Let me investigate and get back to you on that.